tests/components/switcher_kis/test_config_flow.py

```diff
@@ -43,13 +50,90 @@ async def test_user_setup(
         assert mock_bridge.is_running is False
         assert result2["type"] is FlowResultType.CREATE_ENTRY
         assert result2["title"] == "Switcher"
-        assert result2["result"].data == {}
+        assert result2["result"].data == {CONF_USERNAME: None, CONF_TOKEN: None}
 
         await hass.async_block_till_done()
 
         assert len(mock_setup_entry.mock_calls) == 1
 
 
+@pytest.mark.parametrize(
+    "mock_bridge",
+    [
+        [
+            DUMMY_SINGLE_SHUTTER_DUAL_LIGHT_DEVICE,
+        ]
+    ],
+    indirect=True,
+)
+async def test_user_setup_found_token_device(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_bridge
+) -> None:
+    """Test we can finish a config flow with token device found."""
+    with patch("homeassistant.components.switcher_kis.utils.DISCOVERY_TIME_SEC", 0):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+
+        assert result["type"] is FlowResultType.FORM
+        assert result["step_id"] == "confirm"
+
+        result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+
+        assert mock_bridge.is_running is False
+        assert result2["type"] is FlowResultType.FORM
+        assert result2["step_id"] == "credentials"
+
+        with patch("aiohttp.ClientSession.post") as mock_post:
+            # Setup mock response for successful validation
+            mock_response = mock_post.return_value.__aenter__.return_value
+            mock_response.status = 200
+            mock_response.json.return_value = {"result": "true"}
```

Comment:
Please patch the `validate_token` method to return true or false and not the `post` method
```suggestion
        with patch(
            "homeassistant.components.switcher_kis.config_flow.validate_token",
            return_value=True,
        ):
```

Reply:
It is done.